ipcore/hw_tcl_generator.py

```diff
@@ -0,0 +1,268 @@
+
+        self.vendor = dataplane_config.vendor
+
+        self.clock_rate = dataplane_config.clock_rate
+
+        self.data_bus_size = dataplane_config.data_bus_size
+        self.address_bus_size = dataplane_config.address_bus_size
+        self.has_avalon_mm_slave_signal = dataplane_config.has_avalon_mm_slave_signal
+
+        self.sink_max_channel = dataplane_config.sink_max_channel
+        self.source_max_channel = dataplane_config.source_max_channel
+
+    def write_tcl(self, outputFilename):
+        with open(outputFilename, "w") as out_file:
+            out_file.write(self.generate())
+    
+    def generate(self):
+        tcl = self.create_header_file_stuff()
+        tcl += self.create_module()
+        tcl += self.create_file_sets()
+        tcl += self.create_module_assignments()
+        tcl += self.create_connection_point_clock()
+        tcl += self.create_connection_point_reset()
+        tcl += self.create_mm_connection_point()
+        tcl += self.create_sink_connection_point()
+        tcl += self.create_source_connection_point()
+        return tcl
+
+    def create_header_file_stuff(self):
+        tcl = "# # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Built in create_header_file_stuff\n"
+        tcl += "# # # # # # # # # # # # # # # # # #\n\n"
+        tcl += "package require -exact qsys 16.1\n"
+        tcl += "# End create_header_file_stuff\n\n\n"
+        return tcl
+
+
+    def create_module(self):
+        tcl = "# # # # # # # # # # # # # # # # #\n"
+        tcl += "# Created in create_module\n"
+        tcl += "# # # # # # # # # # # # # # # # #\n\n"
+        tcl += "set_module_property DESCRIPTION \"" + \
+            self.description + "\"\n"
+        tcl += "set_module_property NAME \"" + self.name + "\"\n"
+        tcl += "set_module_property VERSION " + self.version + "\n"
+        tcl += "set_module_property OPAQUE_ADDRESS_MAP true\n"
+        tcl += "set_module_property AUTHOR \"" + self.author + "\"\n"
+        tcl += "set_module_property GROUP \"FPGA Open Speech Tools/Autogen\"\n"
+        tcl += "set_module_property DISPLAY_NAME \"" + \
+            self.name + "\"\n"
+        tcl += "set_module_property INSTANTIATE_IN_SYSTEM_MODULE true\n"
+        tcl += "set_module_property EDITABLE true\n"
+        tcl += "set_module_property REPORT_TO_TALKBACK false\n"
+        tcl += "set_module_property ALLOW_GREYBOX_GENERATION false\n"
+        tcl += "set_module_property REPORT_HIERARCHY false\n"
+        tcl += "# end of create_module\n\n\n"
+        return tcl
+
+
+    def create_file_sets(self):
+        tcl = "# # # # # # # # # # # # # # # # # #\n"
+        tcl += "# created in create_file_sets\n"
+        tcl += "# # # # # # # # # # # # # # # # # #\n\n"
+        tcl += "add_fileset QUARTUS_SYNTH QUARTUS_SYNTH \"\" \"\"\n"
+
+        # add additional dependent files before the top level file; quartus needs to synthesize dependencies first
```

Comment:
I think this comment might not actually be true. I don't think the order of the files actually matters.

Reply:
Yeah, I didn't actually notice this comment before but I agree that I don't think it is true